test/coverage/datamodel/PayloadBuilderTest.cpp

```diff
@@ -23,16 +26,667 @@ class TestFixture : public testing::Test {
+	// Arrange
+	uprotocol::v1::UPayloadFormat invalid_format =
+	    static_cast<uprotocol::v1::UPayloadFormat>(
+	        9999);  // Replace with an invalid format
+
+	// Act and Assert
+	EXPECT_THROW(Payload payload(testStringPayload_, invalid_format),
+	             std::out_of_range);
+}
+
+// Create empty string payload and check if it thorws exception
+TEST_F(PayloadTest, EmptyStringPayloadTest) {
+	// Arrange
+	std::string value_string = "";
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+
+	// Act
+	Payload payload(value_string, format);
+
+	// Assert
+	EXPECT_THROW(payload.build(), Payload::PayloadMoved);
+}
+
+// Create std::string payload and call move on payload object test
+TEST_F(PayloadTest, StringMovePayloadTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+
+	// Act
+	Payload payload(testStringPayload_, format);
+	auto [serializedData, payloadFormat] = std::move(payload).move();
+
+	// Assert
+	EXPECT_EQ(serializedData, testStringPayload_);
+	EXPECT_EQ(payloadFormat, format);
+}
+
+// Create string payload and call move() twice to see if it throws exception
+TEST_F(PayloadTest, MoveStringPayloadTwiceTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+
+	// Act
+	Payload payload(testStringPayload_, format);
+	auto movedPayload = std::move(payload).move();
+
+	// Assert
+	EXPECT_EQ(std::get<Payload::PayloadType::Data>(movedPayload),
+	          testStringPayload_);
+	EXPECT_EQ(std::get<Payload::PayloadType::Format>(movedPayload), format);
+
+	// Check that the original payload has been moved
+	EXPECT_THROW(std::move(payload).move(), Payload::PayloadMoved);
+}
+
+// Create string payload and call move() before calling build()
+TEST_F(PayloadTest, MoveStringPayloadBeforeBuildTest) {
+	// Arrange
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+
+	// Act
+	Payload payload(testStringPayload_, format);
+	auto movedPayload = std::move(payload).move();
+
+	// Assert
+	EXPECT_EQ(std::get<Payload::PayloadType::Data>(movedPayload),
+	          testStringPayload_);
+	EXPECT_EQ(std::get<Payload::PayloadType::Format>(movedPayload), format);
+
+	// Check that the original payload has been moved
+	EXPECT_THROW(payload.build(), Payload::PayloadMoved);
+}
+
+/****************RValue String Payload Tests*******************/
+
+// Create RValue String Payload
+TEST_F(PayloadTest, RValueStringPayloadTest) {
+	// Arrange
+	std::string value_string = testStringPayload_;
+	uprotocol::v1::UPayloadFormat format =
+	    uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+	// Act
+	Payload payload(std::move(value_string), format);
+
+	// Assert
+	auto [serializedData, payloadFormat] = payload.build();
+	EXPECT_EQ(serializedData, testStringPayload_);
+	EXPECT_EQ(payloadFormat, format);
+}
```

Comment:
Done. The only way to check value is moved is to check if string is empty. We cannot test the original address because after moving the value, the destination gets new memory address even though move operation was successful.
Following code will not work:
const void *original_address = value_string.data();
//Code
 EXPECT_EQ(serializedData.data(), original_address);

Reply:
Done